src/services/CleanupService.ts

```diff
@@ -363,15 +497,35 @@ export const cleanLocalstorage = (bool?: boolean) => {
 /** Filter the deleted cookies */
 export const filterLocalstorage = (obj: CleanReasonObject, debug = false) => {
   const notProtectedByOpenTab = obj.reason !== ReasonKeep.OpenTabs;
-  const notInAnyLists = (obj.reason === ReasonClean.NoMatchedExpression || obj.reason === ReasonClean.StartupNoMatchedExpression);
+  const notInAnyLists =
+    obj.reason === ReasonClean.NoMatchedExpression ||
+    obj.reason === ReasonClean.StartupNoMatchedExpression;
   const listCleanLocalstorage = cleanLocalstorage(
     obj.expression ? obj.expression.cleanLocalStorage : undefined,
   );
   const nonBlankCookieHostName = obj.cookie.hostname.trim() !== '';
-  cadLog({
-    msg: 'CleanupService.filterLocalstorage: debug data.',
-    x: {notProtectedByOpenTab, notInAnyLists, listCleanLocalstorage, nonBlankCookieHostName, clean1: notInAnyLists, clean2: (notProtectedByOpenTab && listCleanLocalstorage) , CleanReasonObject: obj},
-  }, debug);
+  let cro: CleanReasonObject = obj;
+  if (debug) {
+    // We need deep copying object to as to not change actual cookies
+    const sanitized: CleanReasonObject = JSON.parse(JSON.stringify(obj));
```

Comment:
Can't you use object spread here?

`const sanitized: CleanReasonObject = {...obj};`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
That itself won't do it due it being only shallow copied, which means that the inner object would only be a reference to the actual object, thus would restore cookies with value of `'***'`.

I went one step further and found that this works:
```js
    const sanitized: CleanReasonObject = {
      ...obj,
      cookie: { ...obj.cookie, value: '***' },
    };
```
so I'll adjust accordingly.

Reply:
Should be resolved through 4cb01b4